### ISSUE TITLE
Support sending request trailers

### ISSUE THREAD

**Issue by aajtodd** [square/okhttp#7498]
Add support for sending trailing headers (rfc7230-[4.1](https://www.rfc-editor.org/rfc/rfc7230#section-4.1)/[4.4](https://www.rfc-editor.org/rfc/rfc7230#section-4.4)) in a request.

Currently trailers are supported for [Response](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response/-builder/trailers/) objects but there is no way to set trailers on a request.

**Comment by yschimke** [comment]
Request is reusable in OkHttp.  So Call.setRequestTrailers(Headers)?

**Comment by JakeWharton** [comment]
Seems like the point of trailers is for things discovered while sending the request. Otherwise they would just be normal headers. So it's probably something like a callback that's invoked after the entire body has been written to produce them.

**Comment by swankjesse** [comment]
Presumably useful to checksum the request body?

**Comment by swankjesse** [comment]
Presumably useful to sign or checksum the request body?

**Comment by aajtodd** [comment]
Yeah the primary motivation I'm aware of is checksums. This ended up not being necessary for us (we do chunked encoding with ability to checksum without trailer support).  It would still be good feature for completeness sake.

**Comment by yschimke** [comment]
Given the limited browser support, going to close as not planned.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Trailer#browser_compatibility
